### ISSUE TITLE
[Issue]: Jellyfin can't detect any file in location subfolders

### ISSUE THREAD

**Issue by yoGhurrt1** [jellyfin/jellyfin#7692]
### Please describe your bug

TL;DR - Folder view shows music only in folders that in root directory that I'll choose as "library". Any subfolder in Jellyfin shows as empty.

Hi,

What I tried to achieve - I wanted to use Folder view and Library view side-by-side. Yes, I do understand that Library view is a way to go, but using just a file explorer to manage my music library, it's a few-step process so I would prefer to start in somehow known environment.

My Jellyfin server is on different device so after some digging I figured out that I have to mount network share as hard drive to be visible properly for server. Everything looks fine, I would add this mounted disk to Jellyfin as library, scan for music and in theory be ready to go. Unfortunately, it never worked as it should and my library is incomplete - everything that is in any subfolder, isn't recognized by Jellyfin. 
Problem started with first try when I tried adding library from location "Mounted share:\<username>\Music". After some reaserach, there were posts about Jellyfin that had problems with subfolders deeper then 3rd "level". Ok, with that I mounted this share again but this time directly to this specific folder - same thing.
More reading suggested "Picture" or "Other" mode - it partly worked (after erasing metadata and cache from Jellyfin server), I saw all subfolders but those mode showed only pictures from specific folders. I tried once again with "Music" mode with cleaning Jellyfin configuration but same result. 
Looking into debbug log nothing really looked interesting beside warning about long "-" in files name, that wasn't recognized by Jellyfin but renaming everything in library didn't solve this problem.

What's interesting - I have this problem with web GUI, official program for desktop and Android app but Sonixd (sort of) shows its as it should.

Anyone could help with this?
Thanks in advance.

### Jellyfin Version

10.7.7

### Environment

```markdown
- OS: Windows
- Virtualization: No
- Clients: Windows, Android
- Browser: Firefox 99
- FFmpeg Version:
- Playback Method: Direct Stream
- Hardware Acceleration: No
- Plugins: AudioDB, LDAP-Auth, Last.fm, MusicBrainz and OMDb
- Reverse Proxy: Nginx
- Base URL: none
- Networking: NAT
- Storage: SMB
```

### Please attach any screenshots here
Folder view in "root" folder
![obraz](https://user-images.githubusercontent.com/13383620/166163112-997bc227-c707-4d44-b9a8-b54258057743.png)
View in subfolder with more subfolders inside
![obraz](https://user-images.githubusercontent.com/13383620/166163141-f11c37b6-7523-405c-9ff7-45a8b8d8f8d1.png)
View in subfolder that includes music without any additional subfolders
![obraz](https://user-images.githubusercontent.com/13383620/166163176-9ee7eee3-72e0-45a8-81d6-33c2bf8dd653.png)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Parmenide78** [comment]
+1
I have the same issue.
I have noticed that this issue doesn't exist for movies (just music)
So I wonder if there is a link with the issue number 5517 ? (if it helps)

**Comment by Parmenide78** [comment]
It seems the issue exists also for jellyfin 10.6.4

**Comment by g-battaglia** [comment]
Hi, I have the same issue also for movie
